internal/pkg/service/buffer/storage/compression/reader/reader.go

```diff
@@ -51,9 +51,30 @@ func newParallelGZIPReader(r io.Reader) (io.Reader, error) {
 	return out, nil
 }
 
-func newZstdReader(r io.Reader, cfg compression.Config) (io.Reader, error) {
-	return zstd.NewReader(
+func newZstdReader(r io.Reader, cfg compression.Config) (io.ReadCloser, error) {
+	zstdReader, err := zstd.NewReader(
 		r,
 		zstd.WithDecoderConcurrency(cfg.ZSTD.Concurrency),
 	)
+	if err != nil {
+		return nil, err
+	}
+	return &noErrorCloser{reader: zstdReader}, nil
+}
```

Comment:
Zaroven fixneme drobnu chybu na tomto mieste.

`zstdReader` ma metodu `Close()` nie standardnu `Close() error` :man_shrugging:, dovod bude, ze pri uvolnovani resources nemoze nastat ziadna chyba, ale nebolo to v kode doteraz "typovo" osetrene.
Bez fixu by mohla zostat neuvolnena nejaka pamat (aj ked ak neexistuje ziadna referencia, tak by ju mal v tomto pripade uvolnit GC).

Teda metodu `Close()` prevedieme na standardnu `Close() error` (`io.Closer` interface).